ang/crmCaseType.js

```diff
@@ -300,16 +308,25 @@
             });
           }
         }
-        // TODO The ids below really should use names not labels see
-        //  https://lab.civicrm.org/dev/core/issues/774
+        // TODO The `id` value is caught in between trying to support xml
+        // files where using the relType.id doesn't make it portable between
+        // installs and the xml doesn't have a concept of direction, but
+        // using name_a_b would conflict with it being used currently to set
+        // displaylabel elsewhere. So for now `id` is label.
         else {
           result.push({
+            machineName_a_b: relType.name_a_b,
+            machineName_b_a: relType.name_b_a,
```

Comment:
Ok I see the point. I like xmlName. Not sure about what to call the other one.At the moment I've put `xmlNameOtherDirection`. `ClientName` to me means the client's name, like Steve. `UIName` seems like label. This also touches on some of the other thoughts I'm still collecting, and is the main reason I don't think this PR a perfect fix. The whole thing should be based around `civicrm_relationship_type.id` not names, and only converted to name going to/from xml where the original idea was portability between installs.

Reply:
@demeritcowboy I had initially assumed that relationship type ID would be the best, but that would only really make sense if the whole case type were stored as relational tables rather than XML.  Right now, the case type only lives in XML.

For what to call the other one, how about `systemName` or just `name`?  The name, given from the perspective of the client, is the convention everywhere but in the XML.